src/App.jsx

```diff
@@ -1,37 +1,48 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = React.useState(() => {
-    const savedTodoList = localStorage.getItem("savedTodoList");
-    return savedTodoList ? JSON.parse(savedTodoList) : [];
-  });
-
-  React.useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState([])
+  const [isLoading, setIsLoading] = useState(true)
+
+  useEffect(() => {
+    const fetchData = () => {
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          const savedTodoList = localStorage.getItem("savedTodoList")
+          const initialTodoList = savedTodoList ? JSON.parse(savedTodoList) : []
+          resolve({ data: { todoList: initialTodoList } })
+        }, 2000)
+      });
+    };
+
+    fetchData().then(result => {
```

Comment:
Nicely done wrapping the Promise object in a function, which solved the ambiguity in the instruction! Usually, the syntax chained the .then() method right after calling the Promise constructor. Try it out, and the outcome will end up just the same!